README.md

```diff
@@ -1,3 +1,227 @@
+Wickham et al. ([2016](https://ggplot2-book.org/)) and in the
+documentation page for the function
+[`geom_jitter()`](https://ggplot2.tidyverse.org/reference/geom_jitter.html).
+
+In the context of OD data jittering refers to randomly moving start and
+end points associated with OD pairs, as described in an under review
+paper on the subject (Lovelace et al. under review). The technique is
+implemented in the function
+[`od_jitter()`](https://itsleeds.github.io/od/reference/od_jitter.html)
+in the [`od`](https://itsleeds.github.io/od/index.html) R package. The
+functionality contained in this repo is an extended and much faster
+implementation: according to our benchmarks on a large dataset it was
+around 1000 times faster than the R implementation.
+
+The crate is still a work in progress: the API may change. Issues and
+pull requests are particularly useful at this stage.
+
+<!-- Todo: add worked example. Will do for GISRUK paper -->
+<!-- An illustration of its functionality is shown below. -->
+
+# Installation
+
+Install the package from the system command line as follows (you need to
+have installed and set-up
+[cargo](https://doc.rust-lang.org/cargo/getting-started/installation.html)
+first):
+
+``` bash
+cargo install --git https://github.com/dabreegster/odjitter
+```
+
+To check the package installation worked, you can run `odjitter` command
+without arguments. If it prints the following message congratulations,
+it works 🎉
+
+``` bash
+odjitter
+```
+
+    ## error: The following required arguments were not provided:
+    ##     --od-csv-path <OD_CSV_PATH>
+    ##     --zones-path <ZONES_PATH>
+    ##     --output-path <OUTPUT_PATH>
+    ##     --max-per-od <MAX_PER_OD>
+    ## 
+    ## USAGE:
+    ##     odjitter [OPTIONS] --od-csv-path <OD_CSV_PATH> --zones-path <ZONES_PATH> --output-path <OUTPUT_PATH> --max-per-od <MAX_PER_OD>
+    ## 
+    ## For more information try --help
+
+# Usage
+
+To run algorithm you need a minimum of three inputs, examples of which
+are provided in the `data/` folder of this repo:
+
+1.  A .csv file containing OD data with the first two columns containing
+    zone IDs and subsequent columns representing trip counts:
+
+| geo\_code1 | geo\_code2 | all | from\_home | train | bus | car\_driver | car\_passenger | bicycle | foot | other |
+|:-----------|:-----------|----:|-----------:|------:|----:|------------:|---------------:|--------:|-----:|------:|
+| S02001616  | S02001616  |  82 |          0 |     0 |   3 |           6 |              0 |       2 |   71 |     0 |
+| S02001616  | S02001620  | 188 |          0 |     0 |  42 |          26 |              3 |      11 |  105 |     1 |
+| S02001616  | S02001621  |  99 |          0 |     0 |  13 |           7 |              3 |      15 |   61 |     0 |
+
+1.  A [.geojson
+    file](https://github.com/dabreegster/odjitter/blob/main/data/zones.geojson)
+    representing zones, with the first column matching the zone IDs in
```

Comment:
"with a property (called InterZone by default) matching the zone IDs in the OD data"

GeoJSON even less so has any concept of "first column" -- it's got a JSON map of properties. There's no order

Reply:
:+1: